# Notebook 1: Data Understanding

## 1. Project Title

Investigating the Influence of Regional Fire Activity and Meteorological Conditions on Delhi Air Quality and Developing an Explainable AQI Prediction Framework

## 2. Introduction

Air quality is influenced by a combination of anthropogenic activities, meteorological conditions, and natural events such as biomass burning. To investigate these relationships and develop an explainable Air Quality Index (AQI) prediction framework, this study integrates multiple datasets obtained from different sources.

Before performing any preprocessing, feature engineering, statistical analysis, or machine learning, it is essential to understand the characteristics of each dataset. A thorough understanding of the data ensures that appropriate preprocessing techniques are applied and helps identify potential issues such as missing values, inconsistent data types, duplicate records, and differences in temporal resolution.

This notebook focuses solely on exploring the raw datasets without modifying their original structure. The observations recorded in this notebook will guide the preprocessing and integration steps performed in subsequent notebooks.

## 3. Research Objectives

The objectives of this notebook are to:

- Understand the structure of each dataset used in the study.
- Examine the variables available in each dataset.
- Identify the data types of all variables.
- Assess data quality by identifying missing values and duplicate records.
- Examine the temporal coverage of each dataset.
- Evaluate the suitability of each dataset for the research objectives.
- Document initial observations that will guide subsequent preprocessing and analysis.

## 4. Required Libraries

The following libraries are imported to perform data loading, exploration, and visualization. These libraries are sufficient for understanding the datasets and do not include any machine learning libraries, as modelling is performed in later notebooks.

In [1]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings("ignore")

pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", 100)

## 5. Dataset 1: Air Pollution Dataset

### 5.1 Dataset Introduction

The Air Pollution Dataset contains daily observations of air quality measurements collected from monitoring stations across multiple Indian cities. The dataset includes concentrations of major atmospheric pollutants along with the corresponding Air Quality Index (AQI) and AQI category.

For this research, the dataset serves as the primary source for measuring Delhi's air quality. The AQI will be used as the target variable for prediction, while pollutant concentrations will provide supporting information during exploratory data analysis.

At this stage, the dataset is explored in its original form without performing any filtering or preprocessing.

### 5.2 Loading the Dataset

The dataset is loaded into a pandas DataFrame for further exploration. No modifications are made during this step, ensuring that the original dataset remains unchanged throughout the initial investigation.

In [2]:
pollution_df = pd.read_csv(r"city_day/city_day.csv")

### 5.3 Dataset Overview

The first step in understanding a dataset is to examine its overall dimensions and organization. This includes identifying the number of observations, the number of variables, and reviewing the column names. Additionally, inspecting the first and last few records helps verify that the dataset has been loaded correctly and provides an initial understanding of its contents.

In [3]:
print("Dataset Shape:")
print(pollution_df.shape)

print("\nColumn Names:")
print(pollution_df.columns.tolist())
print()

display(pollution_df.head())

display(pollution_df.tail())

Dataset Shape:
(29531, 16)

Column Names:
['City', 'Date', 'PM2.5', 'PM10', 'NO', 'NO2', 'NOx', 'NH3', 'CO', 'SO2', 'O3', 'Benzene', 'Toluene', 'Xylene', 'AQI', 'AQI_Bucket']



,City,Date,PM2.5,PM10,NO,NO2,NOx,NH3,CO,SO2,O3,Benzene,Toluene,Xylene,AQI,AQI_Bucket
0,Ahmedabad,2015-01-01,NaN,NaN,0.92,18.22,17.15,NaN,0.92,27.64,133.36,0.00,0.02,0.00,NaN,NaN
1,Ahmedabad,2015-01-02,NaN,NaN,0.97,15.69,16.46,NaN,0.97,24.55,34.06,3.68,5.50,3.77,NaN,NaN
2,Ahmedabad,2015-01-03,NaN,NaN,17.40,19.30,29.70,NaN,17.40,29.07,30.70,6.80,16.40,2.25,NaN,NaN
3,Ahmedabad,2015-01-04,NaN,NaN,1.70,18.48,17.97,NaN,1.70,18.59,36.08,4.43,10.14,1.00,NaN,NaN
4,Ahmedabad,2015-01-05,NaN,NaN,22.10,21.42,37.76,NaN,22.10,39.33,39.31,7.01,18.89,2.78,NaN,NaN


,City,Date,PM2.5,PM10,NO,NO2,NOx,NH3,CO,SO2,O3,Benzene,Toluene,Xylene,AQI,AQI_Bucket
29526,Visakhapatnam,2020-06-27,15.02,50.94,7.68,25.06,19.54,12.47,0.47,8.55,23.30,2.24,12.07,0.73,41.0,Good
29527,Visakhapatnam,2020-06-28,24.38,74.09,3.42,26.06,16.53,11.99,0.52,12.72,30.14,0.74,2.21,0.38,70.0,Satisfactory
29528,Visakhapatnam,2020-06-29,22.91,65.73,3.45,29.53,18.33,10.71,0.48,8.42,30.96,0.01,0.01,0.00,68.0,Satisfactory
29529,Visakhapatnam,2020-06-30,16.64,49.97,4.05,29.26,18.80,10.03,0.52,9.84,28.30,0.00,0.00,0.00,54.0,Satisfactory
29530,Visakhapatnam,2020-07-01,15.00,66.00,0.40,26.85,14.05,5.20,0.59,2.10,17.05,NaN,NaN,NaN,50.0,Good


### Findings

The Air Pollution Dataset contains **29,531 observations** and **16 variables**, indicating a large collection of daily air quality records. The dataset includes city information, observation dates, pollutant concentrations, the Air Quality Index (AQI), and AQI categories. The initial inspection confirms that the dataset has been loaded correctly and contains the information required for further analysis.

### 5.4 Dataset Structure

Understanding the internal structure of the dataset is essential before preprocessing. This section examines the data type of each variable, the number of non-null observations, and the overall memory usage of the dataset. These details help identify variables that may require type conversion or additional preprocessing in later stages.

In [4]:
pollution_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29531 entries, 0 to 29530
Data columns (total 16 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   City        29531 non-null  object 
 1   Date        29531 non-null  object 
 2   PM2.5       24933 non-null  float64
 3   PM10        18391 non-null  float64
 4   NO          25949 non-null  float64
 5   NO2         25946 non-null  float64
 6   NOx         25346 non-null  float64
 7   NH3         19203 non-null  float64
 8   CO          27472 non-null  float64
 9   SO2         25677 non-null  float64
 10  O3          25509 non-null  float64
 11  Benzene     23908 non-null  float64
 12  Toluene     21490 non-null  float64
 13  Xylene      11422 non-null  float64
 14  AQI         24850 non-null  float64
 15  AQI_Bucket  24850 non-null  object 
dtypes: float64(13), object(3)
memory usage: 3.6+ MB


### Findings

The dataset consists of **13 numerical variables** and **3 categorical variables**. Most pollutant measurements and the AQI are stored as numerical values, while **City**, **Date**, and **AQI_Bucket** are categorical. Since the **Date** column is currently stored as an object, it will be converted to the datetime format during preprocessing to support time-based analysis.

### 5.5 Data Quality Assessment

The quality of the dataset is evaluated by examining missing values, duplicate records, temporal coverage, and descriptive statistics. Identifying these characteristics at an early stage helps determine the preprocessing steps required before integrating this dataset with the remaining environmental datasets.

### 5.5.1 Missing Values

In [5]:
pollution_df.isnull().sum()

City              0
Date              0
PM2.5          4598
PM10          11140
NO             3582
NO2            3585
NOx            4185
NH3           10328
CO             2059
SO2            3854
O3             4022
Benzene        5623
Toluene        8041
Xylene        18109
AQI            4681
AQI_Bucket     4681
dtype: int64

### 5.5.2 Duplicate Records

In [6]:
pollution_df.duplicated().sum()

np.int64(0)

### 5.5.3 Temporal Coverage

In [7]:
pollution_df["Date"] = pd.to_datetime(pollution_df["Date"])

print("Earliest Date :", pollution_df["Date"].min())
print("Latest Date :", pollution_df["Date"].max())

Earliest Date : 2015-01-01 00:00:00
Latest Date : 2020-07-01 00:00:00


### 5.5.4 Spatial Coverage

In [8]:
print(pollution_df["City"].nunique())

print(pollution_df["City"].unique())

26
['Ahmedabad' 'Aizawl' 'Amaravati' 'Amritsar' 'Bengaluru' 'Bhopal'
 'Brajrajnagar' 'Chandigarh' 'Chennai' 'Coimbatore' 'Delhi' 'Ernakulam'
 'Gurugram' 'Guwahati' 'Hyderabad' 'Jaipur' 'Jorapokhar' 'Kochi' 'Kolkata'
 'Lucknow' 'Mumbai' 'Patna' 'Shillong' 'Talcher' 'Thiruvananthapuram'
 'Visakhapatnam']


### 5.5.5 Descriptive Statistics

In [9]:
pollution_df.describe().T

,count,mean,min,25%,50%,75%,max,std
Date,29531,2018-05-14 05:40:15.807118080,2015-01-01 00:00:00,2017-04-16 00:00:00,2018-08-05 00:00:00,2019-09-03 00:00:00,2020-07-01 00:00:00,NaN
PM2.5,24933.0,67.450578,0.04,28.82,48.57,80.59,949.99,64.661449
PM10,18391.0,118.127103,0.01,56.255,95.68,149.745,1000.0,90.60511
NO,25949.0,17.57473,0.02,5.63,9.89,19.95,390.68,22.785846
NO2,25946.0,28.560659,0.01,11.75,21.69,37.62,362.21,24.474746
NOx,25346.0,32.309123,0.0,12.82,23.52,40.1275,467.63,31.646011
NH3,19203.0,23.483476,0.01,8.58,15.85,30.02,352.89,25.684275
CO,27472.0,2.248598,0.0,0.51,0.89,1.45,175.81,6.962884
SO2,25677.0,14.531977,0.01,5.67,9.16,15.22,193.86,18.133775
O3,25509.0,34.49143,0.01,18.86,30.84,45.57,257.73,21.694928


### Findings

The dataset does not contain any duplicate records, indicating that each observation is unique. However, several variables contain missing values, with **Xylene**, **PM10**, **NH3**, **Toluene**, and **AQI** having the highest number of missing observations. These missing values will be addressed during the preprocessing stage.

The dataset covers the period from **1 January 2015** to **1 July 2020** and includes observations from **26 Indian cities**. Since this research focuses on Delhi, only the Delhi records will be retained for further analysis.

The descriptive statistics show a wide range of pollutant concentrations and AQI values, indicating that the dataset includes both low and high pollution conditions. This variation makes the dataset suitable for exploratory analysis and predictive modelling.

### 5.6 Initial Findings

The Air Pollution Dataset provides daily measurements of major air pollutants along with the corresponding Air Quality Index (AQI), making it the primary dataset for this research. It contains more than **29,000 observations** collected over **five years** from **26 Indian cities**, providing sufficient data for analysing air quality patterns.

Although some variables contain missing values, no duplicate records were found. After filtering the dataset to retain only Delhi observations and handling the missing values, the dataset will be ready for integration with the fire, wind, and weather datasets in the next stage of the research.

## 6. Dataset 2: Fire Dataset

## 6.1 Dataset Introduction

The Fire Dataset contains satellite-based observations of fire incidents detected by the **NASA Fire Information for Resource Management System (FIRMS)**. Each record represents a fire detection captured by satellite sensors and includes information such as the date, geographic location, fire radiative power (FRP), confidence level, and other fire-related attributes.

In this research, the Fire Dataset is used to investigate whether regional fire activity contributes to variations in Delhi's air quality. Fire-related variables derived from this dataset will later be integrated with the air pollution and meteorological datasets to examine their influence on AQI and to develop an explainable prediction model.

At this stage, the dataset is explored in its original form without performing any preprocessing or feature engineering.

### 6.2 Loading the Dataset

The Fire Dataset is loaded into a pandas DataFrame to begin the data understanding process. At this stage, no modifications are made to the dataset. The objective is to preserve the original data while examining its structure, quality, and overall suitability for the research.

In [11]:
fire_df = pd.read_csv("DL_FIRE_SV-C2_774473/fire_archive_SV-C2_774473.csv")

### 6.3 Dataset Overview

The first step in understanding the Fire Dataset is to examine its overall structure and contents. This includes identifying the number of observations, the number of variables, and reviewing the available attributes. Additionally, inspecting the first and last few records provides an initial understanding of how fire incidents are recorded and helps verify that the dataset has been loaded correctly.

In [12]:
print("Dataset Shape:")
print(fire_df.shape)

print("\nColumn Names:")
print(fire_df.columns.tolist())

print()

display(fire_df.head())

display(fire_df.tail())

Dataset Shape:
(562486, 15)

Column Names:
['latitude', 'longitude', 'brightness', 'scan', 'track', 'acq_date', 'acq_time', 'satellite', 'instrument', 'confidence', 'version', 'bright_t31', 'frp', 'daynight', 'type']



,latitude,longitude,brightness,scan,track,acq_date,acq_time,satellite,instrument,confidence,version,bright_t31,frp,daynight,type
0,28.98106,77.89695,330.35,0.50,0.41,2015-01-01,744,SNPP,SNPP,n,2,280.03,3.30,D,0
1,28.84811,77.79187,331.86,0.51,0.41,2015-01-01,744,SNPP,SNPP,n,2,285.87,3.74,D,0
2,29.07977,77.24300,326.44,0.53,0.42,2015-01-01,744,SNPP,SNPP,n,2,281.06,3.85,D,0
3,28.59729,77.92941,326.66,0.50,0.41,2015-01-01,744,SNPP,SNPP,n,2,283.24,3.33,D,0
4,29.94886,76.69364,326.43,0.55,0.43,2015-01-01,744,SNPP,SNPP,n,2,291.31,2.63,D,0


,latitude,longitude,brightness,scan,track,acq_date,acq_time,satellite,instrument,confidence,version,bright_t31,frp,daynight,type
562481,31.47296,75.12141,343.37,0.41,0.45,2020-07-01,827,SNPP,SNPP,n,2,298.26,4.14,D,0
562482,31.85460,74.57639,339.81,0.38,0.43,2020-07-01,827,SNPP,SNPP,n,2,299.72,8.01,D,0
562483,31.60378,74.36993,310.59,0.39,0.36,2020-07-01,2050,SNPP,SNPP,n,2,299.39,1.16,N,2
562484,31.41393,76.83524,303.90,0.43,0.38,2020-07-01,2050,SNPP,SNPP,n,2,292.39,0.89,N,0
562485,29.69294,77.40350,305.97,0.48,0.40,2020-07-01,2051,SNPP,SNPP,n,2,288.59,2.34,N,0


### Findings

The Fire Dataset contains **562,486 observations** and **15 variables**, indicating a large collection of satellite-detected fire events. The available variables include the geographic location of each fire, detection date and time, satellite information, thermal measurements, Fire Radiative Power (FRP), confidence level, and day/night classification.

The first and last records confirm that each row represents an individual fire detection rather than aggregated daily data. This detailed event-level information will support the analysis of regional fire activity and its potential influence on Delhi's air quality.

### 6.4 Dataset Structure

Understanding the internal structure of the Fire Dataset is essential before performing any preprocessing or analysis. This section examines the data type of each variable, the number of non-null observations, and the overall memory usage of the dataset. These details help identify variables that may require type conversion and provide an overview of the dataset's completeness.

In [13]:
fire_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 562486 entries, 0 to 562485
Data columns (total 15 columns):
 #   Column      Non-Null Count   Dtype  
---  ------      --------------   -----  
 0   latitude    562486 non-null  float64
 1   longitude   562486 non-null  float64
 2   brightness  562486 non-null  float64
 3   scan        562486 non-null  float64
 4   track       562486 non-null  float64
 5   acq_date    562486 non-null  object 
 6   acq_time    562486 non-null  int64  
 7   satellite   562486 non-null  object 
 8   instrument  562486 non-null  object 
 9   confidence  562486 non-null  object 
 10  version     562486 non-null  int64  
 11  bright_t31  562486 non-null  float64
 12  frp         562486 non-null  float64
 13  daynight    562486 non-null  object 
 14  type        562486 non-null  int64  
dtypes: float64(7), int64(3), object(5)
memory usage: 64.4+ MB


### Findings

The Fire Dataset contains **15 variables**, including **10 numerical variables** and **5 categorical variables**. Geographic coordinates, thermal measurements, and fire-related attributes are stored as numerical values, while variables such as **acq_date**, **satellite**, **instrument**, **confidence**, and **daynight** are stored as categorical data.

All variables contain **562,486 non-null values**, indicating that the dataset has **no missing values**. The **acq_date** column is currently stored as an object and will be converted to the datetime format during preprocessing to support time-based analysis. The dataset occupies approximately **64.4 MB** of memory, making it suitable for further processing and analysis.

## 6.5 Data Quality Assessment

### 6.5.1 Missing Values

Before integrating the Fire Dataset with the remaining datasets, it is important to examine the presence of missing values. Missing data can affect the quality of feature engineering, statistical analysis, and predictive modelling. This section identifies whether any variables contain missing observations and determines if data cleaning is required.

In [14]:
fire_df.isnull().sum()

latitude      0
longitude     0
brightness    0
scan          0
track         0
acq_date      0
acq_time      0
satellite     0
instrument    0
confidence    0
version       0
bright_t31    0
frp           0
daynight      0
type          0
dtype: int64

### Findings

No missing values were identified in any of the variables of the Fire Dataset. This indicates that the dataset is complete and does not require missing value treatment during the preprocessing stage. The absence of missing data improves the reliability of subsequent feature engineering, statistical analysis, and dataset integration.

### 6.5.2 Duplicate Records

Duplicate records can introduce bias into the analysis by representing the same fire event multiple times. Therefore, it is important to verify whether the Fire Dataset contains any duplicate observations before proceeding with preprocessing and data integration.

In [15]:
fire_df.duplicated().sum()

np.int64(0)

### Findings

No duplicate records were found in the Fire Dataset. This indicates that each observation represents a unique fire detection and no duplicate removal is required during the preprocessing stage.

### 6.5.3 Temporal Coverage

Understanding the temporal coverage of the Fire Dataset is important to determine the period over which fire observations are available. This helps verify whether the dataset aligns with the Air Pollution Dataset and is suitable for time-based analysis and dataset integration.

In [16]:
fire_df["acq_date"] = pd.to_datetime(fire_df["acq_date"])

print("Earliest Date :", fire_df["acq_date"].min())
print("Latest Date   :", fire_df["acq_date"].max())

Earliest Date : 2015-01-01 00:00:00
Latest Date   : 2020-07-01 00:00:00


### Findings

The Fire Dataset covers the period from **1 January 2015** to **1 July 2020**, providing more than five years of fire observations. This temporal coverage matches that of the Air Pollution Dataset, making both datasets suitable for date-based integration and further analysis.

### 6.5.4 Spatial Coverage

The spatial coverage of the Fire Dataset is examined to understand the geographical extent of the recorded fire events. This helps verify whether the dataset includes fire observations from the study region and provides sufficient spatial information for analysing the potential influence of regional fire activity on Delhi's air quality.

In [17]:
print("Latitude Range :")
print("Minimum:", fire_df["latitude"].min())
print("Maximum:", fire_df["latitude"].max())

print("\nLongitude Range :")
print("Minimum:", fire_df["longitude"].min())
print("Maximum:", fire_df["longitude"].max())

Latitude Range :
Minimum: 28.00004
Maximum: 31.99999

Longitude Range :
Minimum: 74.0
Maximum: 77.99998


### Findings

The Fire Dataset covers a geographical area with latitudes ranging from **28.00004° to 31.99999°** and longitudes ranging from **74.00000° to 77.99998°**. These coordinates define the spatial extent of the fire observations included in the dataset.

The available latitude and longitude information provides the geographical location of each fire event, which will be useful for analysing regional fire activity and integrating the dataset with Delhi's air quality data.

### 6.5.5 Descriptive Statistics

Descriptive statistics provide a summary of the numerical variables in the Fire Dataset, including measures such as count, mean, standard deviation, minimum, maximum, and quartiles. These statistics help understand the overall distribution and range of the recorded fire observations before further analysis and preprocessing.

In [18]:
fire_df.describe().T

,count,mean,min,25%,50%,75%,max,std
latitude,562486.0,30.503757,28.00004,30.052412,30.48675,30.974718,31.99999,0.718675
longitude,562486.0,75.531797,74.0,74.87348,75.43922,76.06598,77.99998,0.858313
brightness,562486.0,338.274933,208.0,333.12,337.64,343.34,367.0,10.463672
scan,562486.0,0.45327,0.32,0.39,0.43,0.5,0.8,0.08021
track,562486.0,0.47037,0.36,0.38,0.44,0.55,0.78,0.106073
acq_date,562486,2017-09-18 01:06:56.739972352,2015-01-01 00:00:00,2016-10-16 00:00:00,2017-10-21 00:00:00,2018-11-08 00:00:00,2020-07-01 00:00:00,NaN
acq_time,562486.0,846.396922,647.0,746.0,809.0,834.0,2153.0,260.953726
version,562486.0,2.0,2.0,2.0,2.0,2.0,2.0,0.0
bright_t31,562486.0,301.966502,248.75,298.3,301.49,305.29,362.83,6.992977
frp,562486.0,6.460074,0.09,3.05,4.72,7.59,211.37,6.37419


### Findings

The descriptive statistics indicate that the Fire Dataset contains **562,486** fire observations with complete numerical information for all key variables. The latitude and longitude values confirm a well-defined geographical coverage, while the brightness values range from **208.00 K** to **367.00 K**, with an average of **338.27 K**.

The Fire Radiative Power (FRP), which represents the intensity of detected fires, ranges from **0.09 MW** to **211.37 MW**, with a mean value of **6.46 MW**. Overall, the descriptive statistics provide a comprehensive summary of the numerical characteristics of the dataset and establish a baseline for the detailed exploratory analysis performed in the next notebook.

### 6.6 Initial Findings

The initial assessment of the Fire Dataset leads to the following observations:

- The dataset contains **562,486** satellite-detected fire observations with **15 variables**, providing a comprehensive record of fire events.
- No missing values or duplicate records were identified, indicating good data quality.
- The dataset covers the period from **1 January 2015** to **1 July 2020**, which aligns with the temporal coverage of the Air Pollution Dataset.
- Each fire event includes geographical coordinates, thermal measurements, Fire Radiative Power (FRP), and other fire-related attributes that are valuable for analysing regional fire activity.
- The dataset is well-structured and provides sufficient information for preprocessing, feature engineering, and integration with the Air Pollution and Weather Datasets in the subsequent stages of the project.

## 7. Dataset 3: Wind Dataset

### 7.1 Dataset Introduction

The Wind Dataset contains meteorological information related to wind conditions for the study period. Wind plays a crucial role in the transport and dispersion of air pollutants and smoke generated from fire events. Variables such as wind speed and wind direction help determine whether emissions from surrounding regions can be transported towards Delhi.

This dataset will be integrated with the Air Pollution, Fire, and Weather Datasets in the later stages of the project to investigate the relationship between regional fire activity, meteorological conditions, and Delhi's air quality. In this notebook, the dataset is examined to understand its structure, quality, and overall characteristics before preprocessing.

### 7.2 Loading the Dataset

The Wind Dataset is loaded into a pandas DataFrame to enable further examination of its structure and quality. At this stage, the data is imported without any modifications or preprocessing. The original dataset is preserved to ensure that the subsequent analysis is based on the raw data.

In [21]:
wind_df = pd.read_csv("wind_RAW.csv")

### 7.3 Dataset Overview

The Wind Dataset is examined to understand its overall dimensions, available variables, and sample records. This initial overview helps verify that the dataset has been loaded correctly and provides a basic understanding of the information available for subsequent analysis.

In [22]:
print("Dataset Shape:", wind_df.shape)

print("\nColumn Names:")
print(wind_df.columns.tolist())

print("\nFirst Five Records:")
display(wind_df.head())

print("\nLast Five Records:")
display(wind_df.tail())

Dataset Shape: (2533952, 9)

Column Names:
['time', 'latitude', 'longitude', 'number', 'step', 'surface', 'valid_time', 'u10', 'v10']

First Five Records:


,time,latitude,longitude,number,step,surface,valid_time,u10,v10
0,2015-01-01 00:00:00,32.0,74.00,0,0 days,0.0,2015-01-01 00:00:00,-0.831787,-1.742523
1,2015-01-01 00:00:00,32.0,74.25,0,0 days,0.0,2015-01-01 00:00:00,-0.656982,-1.855804
2,2015-01-01 00:00:00,32.0,74.50,0,0 days,0.0,2015-01-01 00:00:00,-0.967529,-2.244476
3,2015-01-01 00:00:00,32.0,74.75,0,0 days,0.0,2015-01-01 00:00:00,-1.294678,-1.374359
4,2015-01-01 00:00:00,32.0,75.00,0,0 days,0.0,2015-01-01 00:00:00,-1.279053,-0.898773



Last Five Records:


,time,latitude,longitude,number,step,surface,valid_time,u10,v10
2533947,2020-12-31 18:00:00,28.0,77.00,0,0 days,0.0,2020-12-31 18:00:00,1.927811,-0.354599
2533948,2020-12-31 18:00:00,28.0,77.25,0,0 days,0.0,2020-12-31 18:00:00,1.950272,-0.468857
2533949,2020-12-31 18:00:00,28.0,77.50,0,0 days,0.0,2020-12-31 18:00:00,2.612381,-0.439560
2533950,2020-12-31 18:00:00,28.0,77.75,0,0 days,0.0,2020-12-31 18:00:00,2.717850,0.045792
2533951,2020-12-31 18:00:00,28.0,78.00,0,0 days,0.0,2020-12-31 18:00:00,2.511795,0.489151


### Findings

The Wind Dataset contains **2,533,952 observations** and **9 variables**, indicating a large collection of wind measurements recorded over multiple locations and time periods. The available variables include temporal information, geographical coordinates, and the **u10** and **v10** wind components, which represent the east-west and north-south wind directions, respectively.

The first and last records confirm that the dataset stores wind measurements for different geographical locations at different timestamps. This detailed spatio-temporal information will support the analysis of wind patterns and their potential role in transporting pollutants and smoke towards Delhi.

### 7.4 Dataset Structure

The structure of the Wind Dataset is examined to understand the data types, non-null values, and memory usage of each variable. This assessment helps identify whether the variables are stored in appropriate formats and provides an overview of the dataset before evaluating its quality and performing preprocessing.

In [23]:
wind_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2533952 entries, 0 to 2533951
Data columns (total 9 columns):
 #   Column      Dtype  
---  ------      -----  
 0   time        object 
 1   latitude    float64
 2   longitude   float64
 3   number      int64  
 4   step        object 
 5   surface     float64
 6   valid_time  object 
 7   u10         float64
 8   v10         float64
dtypes: float64(5), int64(1), object(3)
memory usage: 174.0+ MB


### Findings

The Wind Dataset contains **9 variables**, including **5 numerical variables**, **1 integer variable**, and **3 object variables**. The numerical variables represent geographical coordinates, surface information, and the **u10** and **v10** wind components, while the object variables store temporal information.

The **time** and **valid_time** columns are currently stored as object data types and will be converted to the datetime format during preprocessing to support time-based analysis. The dataset occupies approximately **174 MB** of memory, reflecting its large size and detailed spatio-temporal coverage.

## 7.5 Data Quality Assessment

### 7.5.1 Missing Values

Missing values can affect the accuracy of data analysis, feature engineering, and dataset integration. Therefore, the Wind Dataset is examined to identify whether any variables contain missing observations that may require treatment during the preprocessing stage.

In [25]:
wind_df.isnull().sum()

time          0
latitude      0
longitude     0
number        0
step          0
surface       0
valid_time    0
u10           0
v10           0
dtype: int64

### Findings

No missing values were identified in any of the variables of the Wind Dataset. This indicates that the dataset is complete and does not require missing value treatment during the preprocessing stage. The absence of missing data supports reliable feature engineering, time-based analysis, and integration with the Air Pollution, Fire, and Weather Datasets.

### 7.5.2 Duplicate Records

Duplicate records can introduce redundancy and affect the accuracy of statistical analysis and dataset integration. Therefore, the Wind Dataset is examined to determine whether multiple identical observations exist and whether duplicate removal is required before preprocessing.

In [26]:
wind_df.duplicated().sum()

np.int64(0)

### Findings

No duplicate records were found in the Wind Dataset. This indicates that each observation represents a unique wind measurement, and no duplicate removal is required during the preprocessing stage.

### 7.5.3 Temporal Coverage

The temporal coverage of the Wind Dataset is examined to determine the period over which wind measurements are available. This helps verify whether the dataset aligns with the other datasets used in the study and supports time-based integration and analysis.

In [27]:
wind_df["time"] = pd.to_datetime(wind_df["time"])

print("Earliest Date :", wind_df["time"].min())
print("Latest Date   :", wind_df["time"].max())

Earliest Date : 2015-01-01 00:00:00
Latest Date   : 2020-12-31 18:00:00


### Findings

The Wind Dataset covers the period from **1 January 2015** to **31 December 2020**, providing six years of wind measurements. This temporal coverage includes the entire study period of the Air Pollution and Fire Datasets, ensuring that wind information is available for all dates used in the subsequent integration and analysis.

### 7.5.4 Spatial Coverage

The spatial coverage of the Wind Dataset is examined to understand the geographical extent of the recorded wind measurements. This helps verify whether the dataset provides sufficient spatial information for analysing wind patterns and their potential influence on the transport of pollutants and smoke towards Delhi.

In [28]:
print("Latitude Range:")
print("Minimum:", wind_df["latitude"].min())
print("Maximum:", wind_df["latitude"].max())

print("\nLongitude Range:")
print("Minimum:", wind_df["longitude"].min())
print("Maximum:", wind_df["longitude"].max())

Latitude Range:
Minimum: 28.0
Maximum: 32.0

Longitude Range:
Minimum: 74.0
Maximum: 78.0


### Findings

The Wind Dataset covers a geographical area with latitudes ranging from **28.0° to 32.0°** and longitudes ranging from **74.0° to 78.0°**. These coordinates define the spatial extent of the wind measurements included in the dataset.

The available latitude and longitude information provides spatially distributed wind observations, which will support the analysis of wind patterns and their integration with the Air Pollution and Fire Datasets.

### 7.5.5 Descriptive Statistics

Descriptive statistics provide a summary of the numerical variables in the Wind Dataset, including measures such as count, mean, standard deviation, minimum, maximum, and quartiles. These statistics help understand the overall distribution and range of the recorded wind measurements before further analysis and preprocessing.

In [29]:
wind_df.describe().T

,count,mean,min,25%,50%,75%,max,std
time,2533952,2017-12-31 21:00:00.000000512,2015-01-01 00:00:00,2016-07-01 22:30:00,2017-12-31 21:00:00,2019-07-02 19:30:00,2020-12-31 18:00:00,NaN
latitude,2533952.0,30.0,28.0,29.0,30.0,31.0,32.0,1.224745
longitude,2533952.0,76.0,74.0,75.0,76.0,77.0,78.0,1.224745
number,2533952.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
surface,2533952.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
u10,2533952.0,0.078188,-12.106613,-1.274338,0.080589,1.449936,11.608795,1.938516
v10,2533952.0,-0.325748,-10.495987,-1.390659,-0.462814,0.669464,11.143829,1.535812


### Findings

The descriptive statistics indicate that the Wind Dataset contains **2,533,952** wind measurements with complete numerical information for all key variables. The latitude and longitude values confirm the geographical extent of the recorded observations, while the **u10** and **v10** variables represent the horizontal wind components in the east-west and north-south directions, respectively.

The **u10** values range from **−12.11 m/s** to **11.61 m/s**, while the **v10** values range from **−10.50 m/s** to **11.14 m/s**, indicating that wind measurements capture variations in both magnitude and direction over the study period. Overall, the descriptive statistics provide a comprehensive summary of the numerical characteristics of the dataset and establish a baseline for the detailed exploratory analysis performed in the next notebook.

### 7.6 Initial Findings

The initial assessment of the Wind Dataset leads to the following observations:

- The dataset contains **2,533,952** wind measurements with **9 variables**, providing a comprehensive record of wind conditions over multiple locations and time periods.
- No missing values or duplicate records were identified, indicating good data quality.
- The dataset covers the period from **1 January 2015** to **31 December 2020**, fully encompassing the study period of the Air Pollution and Fire Datasets.
- The dataset includes geographical coordinates along with the **u10** and **v10** wind components, providing the information required to analyse wind patterns and their potential influence on pollutant transport.
- The dataset is well-structured and provides sufficient information for preprocessing, feature engineering, and integration with the Air Pollution, Fire, and Weather Datasets in the subsequent stages of the project.

## 8. Dataset 4: Weather Dataset

### 8.1 Dataset Introduction

The Weather Dataset contains meteorological variables that describe the atmospheric conditions during the study period. Weather factors such as temperature, humidity, precipitation, and surface pressure influence the formation, dispersion, and accumulation of air pollutants. These variables provide important environmental context for understanding variations in Delhi's air quality.

This dataset will be integrated with the Air Pollution, Fire, and Wind Datasets in the later stages of the project to investigate the combined influence of meteorological conditions and regional fire activity on air pollution. In this notebook, the dataset is examined to understand its structure, quality, and overall characteristics before preprocessing.

### 8.2 Loading the Dataset

The Weather Dataset is loaded into a pandas DataFrame to enable further examination of its structure and quality. At this stage, the data is imported without any modifications or preprocessing. The original dataset is preserved to ensure that the subsequent analysis is based on the raw data.

In [31]:
weather_df = pd.read_csv("weather_RAW.csv")

### 8.3 Dataset Overview

The Weather Dataset is examined to understand its overall dimensions, available variables, and sample records. This initial overview helps verify that the dataset has been loaded correctly and provides a basic understanding of the information available for subsequent analysis.

In [32]:
print("Dataset Shape:", weather_df.shape)

print("\nColumn Names:")
print(weather_df.columns.tolist())

print("\nFirst Five Records:")
display(weather_df.head())

print("\nLast Five Records:")
display(weather_df.tail())

Dataset Shape: (2955892, 9)

Column Names:
['time', 'latitude', 'longitude', 'number', 'step', 'surface', 'valid_time', 'd2m', 't2m']

First Five Records:


,time,latitude,longitude,number,step,surface,valid_time,d2m,t2m
0,2015-01-01 00:00:00,32.0,74.00,0,0 days,0.0,2015-01-01 00:00:00,279.61612,279.87158
1,2015-01-01 00:00:00,32.0,74.25,0,0 days,0.0,2015-01-01 00:00:00,279.54190,280.00635
2,2015-01-01 00:00:00,32.0,74.50,0,0 days,0.0,2015-01-01 00:00:00,280.18057,280.45752
3,2015-01-01 00:00:00,32.0,74.75,0,0 days,0.0,2015-01-01 00:00:00,280.68057,281.25440
4,2015-01-01 00:00:00,32.0,75.00,0,0 days,0.0,2015-01-01 00:00:00,280.62198,281.27002



Last Five Records:


,time,latitude,longitude,number,step,surface,valid_time,d2m,t2m
2955887,2022-12-31 18:00:00,28.0,77.00,0,0 days,0.0,2022-12-31 18:00:00,281.36017,283.54382
2955888,2022-12-31 18:00:00,28.0,77.25,0,0 days,0.0,2022-12-31 18:00:00,281.50470,283.52820
2955889,2022-12-31 18:00:00,28.0,77.50,0,0 days,0.0,2022-12-31 18:00:00,281.72150,283.58093
2955890,2022-12-31 18:00:00,28.0,77.75,0,0 days,0.0,2022-12-31 18:00:00,281.85626,283.61414
2955891,2022-12-31 18:00:00,28.0,78.00,0,0 days,0.0,2022-12-31 18:00:00,281.94220,283.55360


### Findings

The Weather Dataset contains **2,955,892 observations** and **9 variables**, indicating a large collection of meteorological measurements recorded over multiple locations and time periods. The available variables include temporal information, geographical coordinates, and the **d2m** (2-metre dew point temperature) and **t2m** (2-metre air temperature) variables, which describe important atmospheric conditions.

The first and last records confirm that the dataset stores weather measurements for different geographical locations at different timestamps. This detailed spatio-temporal information will support the analysis of meteorological conditions and their relationship with Delhi's air quality.

### 8.4 Dataset Structure

The structure of the Weather Dataset is examined to understand the data types, non-null values, and memory usage of each variable. This assessment helps identify whether the variables are stored in appropriate formats and provides an overview of the dataset before evaluating its quality and performing preprocessing.

In [33]:
weather_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2955892 entries, 0 to 2955891
Data columns (total 9 columns):
 #   Column      Dtype  
---  ------      -----  
 0   time        object 
 1   latitude    float64
 2   longitude   float64
 3   number      int64  
 4   step        object 
 5   surface     float64
 6   valid_time  object 
 7   d2m         float64
 8   t2m         float64
dtypes: float64(5), int64(1), object(3)
memory usage: 203.0+ MB


### Findings

The Weather Dataset consists of **9 variables**, including **5 floating-point variables**, **1 integer variable**, and **3 object-type variables**. The **time** and **valid_time** variables are currently stored as object data types and will be converted to datetime format during preprocessing to facilitate time-based analysis.

The dataset occupies approximately **203 MB** of memory, reflecting its large size and extensive meteorological observations collected over multiple locations and time periods.

## 8.5 Data Quality Assessment

### 8.5.1 Missing Values

Missing values can reduce data quality and affect the reliability of statistical analysis and predictive modeling. Therefore, each variable is examined to identify the presence of missing observations before any preprocessing is performed.

In [35]:
weather_df.isnull().sum()

time          0
latitude      0
longitude     0
number        0
step          0
surface       0
valid_time    0
d2m           0
t2m           0
dtype: int64

### Findings

No missing values were identified in any of the variables within the Weather Dataset. This indicates that the dataset is complete and does not require missing value treatment before preprocessing or subsequent analysis. The absence of missing observations supports reliable integration with the other datasets used in this study.

### 8.5.2 Duplicate Records

Duplicate records can introduce bias and affect the accuracy of statistical analysis. Therefore, the dataset is examined to identify whether any duplicate observations are present before preprocessing.

In [37]:
weather_df.duplicated().sum()

np.int64(0)

### Findings

No duplicate records were identified in the Weather Dataset. This indicates that each observation represents a unique weather measurement, and no duplicate removal is required before preprocessing or further analysis.

### 8.5.3 Temporal Coverage

The temporal coverage of the Weather Dataset is examined to determine the study period represented by the weather observations. Understanding the time span of the dataset is essential for assessing its compatibility with the other datasets and supporting time-based integration during subsequent analysis.

In [38]:
weather_df["time"] = pd.to_datetime(weather_df["time"])

print("Earliest Date:", weather_df["time"].min())
print("Latest Date:", weather_df["time"].max())

Earliest Date: 2015-01-01 00:00:00
Latest Date: 2022-12-31 18:00:00


### Findings

The Weather Dataset covers the period from **1 January 2015** to **31 December 2022**, providing nearly eight years of meteorological observations. This time span fully encompasses the study period of the Air Pollution, Fire, and Wind Datasets, making it suitable for time-based integration and comparative analysis in the later stages of the project.

### 8.5.4 Spatial Coverage

The geographical extent of the Weather Dataset is examined using the minimum and maximum latitude and longitude values. Understanding the spatial coverage helps determine the region represented by the weather observations and supports spatial integration with the other datasets.

In [39]:
print("Latitude Range:")
print("Minimum:", weather_df["latitude"].min())
print("Maximum:", weather_df["latitude"].max())

print("\nLongitude Range:")
print("Minimum:", weather_df["longitude"].min())
print("Maximum:", weather_df["longitude"].max())

Latitude Range:
Minimum: 28.0
Maximum: 32.0

Longitude Range:
Minimum: 74.0
Maximum: 78.0


### Findings

The Weather Dataset covers a geographical region bounded by latitudes **28.0° to 32.0°** and longitudes **74.0° to 78.0°**. These coordinates define the spatial extent of the meteorological observations and provide weather information across multiple locations within the study region. The spatial coverage supports integration with the Air Pollution, Fire, and Wind Datasets for subsequent spatial and temporal analyses.

### 8.5.5 Descriptive Statistics

Descriptive statistics are computed for the numerical variables in the Weather Dataset to summarize their central tendency, variability, and overall distribution. This provides an initial understanding of the meteorological measurements before conducting detailed exploratory data analysis.

In [40]:
weather_df.describe().T

,count,mean,min,25%,50%,75%,max,std
time,2955892,2018-08-23 11:26:59.084865536,2015-01-01 00:00:00,2016-10-01 04:30:00,2018-07-02 09:00:00,2020-04-01 13:30:00,2022-12-31 18:00:00,NaN
latitude,2955892.0,30.0,28.0,29.0,30.0,31.0,32.0,1.224745
longitude,2955892.0,76.0,74.0,75.0,76.0,77.0,78.0,1.224745
number,2955892.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
surface,2955892.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
d2m,2955892.0,287.954656,222.77634,282.92017,287.804095,294.96167,303.16675,8.50049
t2m,2955892.0,296.518346,236.58138,289.680138,298.1526,303.87915,322.02026,10.083979


### Findings

The descriptive statistics indicate that the Weather Dataset contains complete numerical information for the geographical coordinates and meteorological variables. The **2-metre dew point temperature (d2m)** ranges from **222.78 K** to **303.17 K**, with an average of approximately **287.95 K**, while the **2-metre air temperature (t2m)** ranges from **236.58 K** to **322.02 K**, with an average of approximately **296.52 K**.

These summary statistics provide an initial understanding of the meteorological conditions represented in the dataset and establish a baseline for the detailed exploratory data analysis performed in the subsequent notebook.

### 8.6 Initial Findings

The initial examination of the Weather Dataset indicates that it is well-structured and suitable for subsequent preprocessing and analysis. The key findings are summarized below:

- The dataset contains **2,955,892 observations** and **9 variables** representing meteorological measurements.
- No missing values or duplicate records were identified, indicating good data quality.
- The dataset covers the period from **1 January 2015** to **31 December 2022**, fully encompassing the study period of the Air Pollution, Fire, and Wind Datasets.
- The geographical coverage spans latitudes **28.0°–32.0°** and longitudes **74.0°–78.0°**, providing weather observations across the study region.
- The dataset includes key meteorological variables such as **2-metre dew point temperature (d2m)** and **2-metre air temperature (t2m)**, which will support the analysis of atmospheric conditions influencing air quality.
- Overall, the dataset is complete and ready for preprocessing and integration with the other datasets in the subsequent stages of the project.

## 9. Comparative Summary of Datasets

The four datasets examined in this notebook provide complementary information required to investigate the relationship between meteorological conditions, regional fire activity, and air quality in Delhi. Each dataset contributes a distinct aspect of the overall analysis while sharing compatible temporal and spatial characteristics that support their integration.

The **Air Pollution Dataset** serves as the primary dataset by providing daily air quality measurements and the target variables for subsequent analysis and prediction. The **Fire Dataset** captures satellite-detected fire events that may contribute to changes in air pollution levels. The **Wind Dataset** provides wind measurements that influence the transport and dispersion of pollutants and smoke, while the **Weather Dataset** describes atmospheric conditions through meteorological variables such as air temperature and dew point temperature.

Across all four datasets, no missing values or duplicate records were identified, indicating good data quality. Furthermore, the temporal coverage of the Weather Dataset fully encompasses the study period of the Air Pollution, Fire, and Wind Datasets, enabling consistent time-based integration. Together, these datasets provide a comprehensive foundation for preprocessing, feature engineering, exploratory data analysis, and predictive modeling in the subsequent stages of the project.

## 10. Conclusion

This notebook presented a comprehensive understanding of the four datasets used in this study, including the Air Pollution, Fire, Wind, and Weather Datasets. Each dataset was examined to understand its structure, variables, data quality, temporal coverage, and spatial coverage before any preprocessing was performed.

The assessment confirmed that all datasets are well-structured, contain no missing values or duplicate records, and possess compatible temporal and spatial characteristics that support their integration. The individual roles of each dataset in representing air quality, fire activity, wind conditions, and meteorological factors were also established.

Overall, the data understanding process provides a strong foundation for the subsequent stages of the project. The next notebook focuses on data preprocessing, where the datasets will be cleaned, transformed, integrated, and prepared for exploratory data analysis and predictive modeling.

## 11. Next Steps

With a clear understanding of the structure, quality, and characteristics of the four datasets, the next phase of the project focuses on **data preprocessing**. The upcoming notebook will prepare the datasets for exploratory data analysis and predictive modeling through the following tasks:

- Converting variables to appropriate data types.
- Cleaning and transforming individual datasets where required.
- Aggregating the Fire, Wind, and Weather Datasets to daily observations.
- Integrating all datasets into a unified analytical dataset based on the date.
- Creating additional temporal and domain-specific features where appropriate.
- Performing final data quality checks before exploratory data analysis.

The processed dataset generated from the preprocessing stage will serve as the foundation for exploratory data analysis, feature engineering, machine learning model development, and interpretation in the subsequent notebooks.